nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -721,10 +722,30 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
                             try {
                                 DataType targetDataType = DataTypeUtils.getDataTypeFromSQLTypeValue(sqlType);
                                 if (targetDataType != null) {
-                                    currentValue = DataTypeUtils.convertType(
-                                            currentValue,
-                                            targetDataType,
-                                            fieldName);
+                                    if (sqlType == Types.BLOB || sqlType == Types.BINARY) {
+                                        if (currentValue instanceof Object[]) {
+                                            // Convert Object[] arrays to byte[]
+                                            Object[] src = (Object[]) currentValue;
+                                            byte[] dest = new byte[src.length];
+                                            for (int j = 0; j < src.length; j++) {
+                                                dest[j] = (byte) src[j];
+                                            }
+                                            currentValue = dest;
+                                        } else if (currentValue instanceof Byte[]) {
```

Comment:
Based on the logic, will this block be exercised since the initial check handles `Object[]`?  Running the unit test in debug mode results in a `currentValue` of `Byte[]` hitting the above conditional and not this conditional.  At minimum, it seems like the ordering should be adjusted to check for `Byte[]` before `Object[]`.

Reply:
Good point, will exchange the order

Reply:
Since Byte[] covers the actual use case, then due to the above comment I really should only have to check for null or byte[] since those will work downstream. Otherwise we don't know how to do the conversion so an exception should be thrown. Agreed?

Reply:
That sounds good, it avoids potentially unexpected behavior, and if the scenario arises, implementing a new check for a more specific array object type seems like the way to go.